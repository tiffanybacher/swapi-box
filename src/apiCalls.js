export const getRandomFilm = (ranNum) => {
  const url = `https://swapi.co/api/films/${ranNum}/`;

  return fetch(url)
    .then(response => {
      if (!response.ok) {
        throw Error('Film failed to load');
      } else {
        return response.json();
      }
    });
}

export const getCardData = (category) => {
  const url = `https://swapi.co/api/${category}/`;

  return fetch(url)
    .then(response => {
      if (!response.ok) {
        throw Error('Card data failed to load');
      } else {
        return response.json();
      }
    });
}

export const getSubData = (url) => {
  return fetch(url)
    .then(response => {
      if (!response.ok) {
        throw Error('Data failed to load');
      } else {
        return response.json();
      }
    });
}