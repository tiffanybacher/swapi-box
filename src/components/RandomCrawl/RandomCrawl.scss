.RandomCrawl {
  height: 70vh;
}