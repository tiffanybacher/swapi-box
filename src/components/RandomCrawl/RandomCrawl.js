import React, { Component } from 'react';
import Crawl from 'react-star-wars-crawl';
import { getRandomFilm } from '../../apiCalls';

class RandomCrawl extends Component {
  constructor() {
    super();
    this.state = {
      film: {}
    }
  }

  componentDidMount = () => {
    const ranNum = Math.floor(Math.random() * 7 + 1);

    return getRandomFilm(ranNum)
      .then(film => this.setState({ film }))
      .catch(error => console.log(error));
  }

  render() {
    const { film } = this.state;
    const romanNumerals = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX'];
    const numeralIndex = film.episode_id - 1;
   
    return (
      <Crawl
        title={`Episode ${romanNumerals[numeralIndex]}`}
        subTitle={film.title}
        text={film.opening_crawl}
      />
    );
  }
}

// "It is a period of civil war. Rebel spaceships, striking from a hidden base, have won their first victory against the evil Galactic Empire. During the battle, Rebel spies managed to steal secret plans to the Empire’s ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet. Pursued by the Empire’s sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy…"

export default RandomCrawl;