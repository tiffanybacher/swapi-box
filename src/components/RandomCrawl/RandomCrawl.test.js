import React from 'react';
import { shallow } from 'enzyme';
import RandomCrawl from './RandomCrawl';
import { getRandomFilm } from '../../apiCalls'
import mockFilmResponse from '../../apiMockData';

describe('RandomCrawl', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow ( <RandomCrawl /> );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({ film: {} });
  });

  describe('componentDidMount', () => {
    beforeAll(() => {
      getRandomFilm.mockImplementation(() => {
        return Promise.resolve(mockFilmResponse);
      });
    });

    it.skip('should invoke getRandomFilm on mount', async () => {
      const getRandomFilm = jest.fn();

      await wrapper.instance().componentDidMount();

      expect(getRandomFilm).toHaveBeenCalled();
    });

    it.skip('should set state to film response on mount', async () => {
      expect(wrapper.state()).toEqual({ film: {} });

      await wrapper.instance().componentDidMount();

      expect(wrapper.state()).toEqual({ film: mockFilmResponse });
    });

    it.skip('should set state with an error if fetch fails', async () => {
      getRandomFilm.mockImplementation(() => {
        return Promise.reject();
      });

      await wrapper.instance.componentDidMount();

      expect(wrapper.state('errorStatus')).toEqual('Error adding film');
    });
  });
});