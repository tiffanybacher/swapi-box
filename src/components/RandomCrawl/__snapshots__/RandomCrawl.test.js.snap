// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomCrawl should match the snapshot 1`] = `
<Crawl
  title="Episode undefined"
/>
`;
